x/clp/abci_test.go

```diff
@@ -223,3 +223,202 @@ func TestBeginBlocker(t *testing.T) {
+		poolAssetPermissions   []tokenregistrytypes.Permission
+		nativeAssetPermissions []tokenregistrytypes.Permission
+		params                 types.Params
+		epoch                  types.PmtpEpoch
+		expectedStates         ExpectedStates
+		err                    error
+		errString              error
+	}{
+		{
+			name:                   "default",
+			createBalance:          true,
+			createPool:             true,
+			createLPs:              true,
+			poolAsset:              "eth",
+			address:                "sif1syavy2npfyt9tcncdtsdzf7kny9lh777yqc2nd",
+			nativeBalance:          sdk.Int(sdk.NewUintFromString(types.PoolThrehold)),
+			externalBalance:        sdk.Int(sdk.NewUintFromString(types.PoolThrehold)),
+			nativeAssetAmount:      sdk.NewUint(1000),
+			externalAssetAmount:    sdk.NewUint(1000),
+			poolUnits:              sdk.NewUint(1000),
+			poolAssetPermissions:   []tokenregistrytypes.Permission{tokenregistrytypes.Permission_CLP},
+			nativeAssetPermissions: []tokenregistrytypes.Permission{tokenregistrytypes.Permission_CLP},
+			params: types.Params{
+				MinCreatePoolThreshold:   types.DefaultMinCreatePoolThreshold,
+				PmtpPeriodGovernanceRate: sdk.MustNewDecFromStr("0.10"),
+				PmtpPeriodEpochLength:    1,
+				PmtpPeriodStartBlock:     1,
+				PmtpPeriodEndBlock:       40,
+			},
+			epoch: types.PmtpEpoch{
+				EpochCounter: 0,
+				BlockCounter: 0,
+			},
+			expectedStates: ExpectedStates{
+				{
+					pool: types.Pool{
+						ExternalAsset:        &types.Asset{Symbol: "eth"},
+						NativeAssetBalance:   sdk.NewUint(1000),
+						ExternalAssetBalance: sdk.NewUint(1000),
+						PoolUnits:            sdk.NewUint(1000),
+					},
+					SwapPriceNative:   sdk.MustNewDecFromStr("0.998002996005000000"),
+					SwapPriceExternal: sdk.MustNewDecFromStr("0.998002996005000000"),
+					pmtpRateParams: types.PmtpRateParams{
+						PmtpPeriodBlockRate:    sdk.NewDec(0),
+						PmtpCurrentRunningRate: sdk.NewDec(0),
+					},
+				},
+				{
+					pool: types.Pool{
+						ExternalAsset:        &types.Asset{Symbol: "eth"},
+						NativeAssetBalance:   sdk.NewUint(1000),
+						ExternalAssetBalance: sdk.NewUint(1000),
+						PoolUnits:            sdk.NewUint(1000),
+					},
+					SwapPriceNative:   sdk.MustNewDecFromStr("0.998002996005000000"),
+					SwapPriceExternal: sdk.MustNewDecFromStr("0.998002996005000000"),
+					pmtpRateParams: types.PmtpRateParams{
+						PmtpPeriodBlockRate:    sdk.NewDec(0),
+						PmtpCurrentRunningRate: sdk.NewDec(0),
+					},
+				},
+				{
+					pool: types.Pool{
+						ExternalAsset:        &types.Asset{Symbol: "eth"},
+						NativeAssetBalance:   sdk.NewUint(1000),
+						ExternalAssetBalance: sdk.NewUint(1000),
+						PoolUnits:            sdk.NewUint(1000),
+					},
+					SwapPriceNative:   sdk.MustNewDecFromStr("0.998002996005000000"),
+					SwapPriceExternal: sdk.MustNewDecFromStr("0.998002996005000000"),
+					pmtpRateParams: types.PmtpRateParams{
+						PmtpPeriodBlockRate:    sdk.NewDec(0),
```

Comment:
Should'nt the block and running rate start changing at height 1 ?

Reply:
@Jedi2002 I am just initiating them here at a nil value but I think I was doing that before we solve the bugs we did last days, will try to get rid of that

Reply:
oh no sorry those are expected values, I just put the values that the current version was returning but now as I am incrementing the height it looks like those values are changing will update the branch with the new values soon